lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -1356,8 +994,7 @@ evaluateAndCacheCall(SILFunction &fn, SubstitutionList substitutions,
 // ConstExprEvaluator implementation.
 //===----------------------------------------------------------------------===//
 
-ConstExprEvaluator::ConstExprEvaluator(SILModule &m)
-  : silLoader(SerializedSILLoader::create(m.getASTContext(), &m, nullptr)) {
+ConstExprEvaluator::ConstExprEvaluator(SILModule &m) {
```

Comment:
Why is this initializer empty?
Should `ConstExprEvaluator.allocator` be set to [`SILModule.BPA`](https://github.com/apple/swift/blob/d365c153d4c762147b75250ca3eefd684602e332/include/swift/SIL/SILModule.h#L132)?

Reply:
The constexpr evaluator intentionally has its own bump pointer allocator, because it generates lots of intermediate garbage and the client (e.g. #assert) only cares about the result.  There is no reason to make that intermediate garbage be immortal.

Reply:
Gotcha, that makes sense. LGTM.